tools/cargo-each/src/combinations_iter/mod.rs

```diff
@@ -0,0 +1,306 @@
+use std::{collections::BTreeSet, iter};
+
+use anyhow::Error;
+use cargo_metadata::Package;
+
+use crate::{
+    check,
+    config::{Combination, Config, Set},
+    either_iter::EitherIter,
+    iter_or_else_iter::IterOrElseIter,
+};
+
+#[cfg(test)]
+mod tests;
+
+pub(crate) fn package_combinations<'r>(
+    package: &'r Package,
+    maybe_config: Option<&'r Config<'r>>,
+    groups: Option<&'r BTreeSet<&'r str>>,
+) -> Result<Option<impl Iterator<Item = String> + 'r>, ()> {
+    if let Some(config) = maybe_config {
+        check::configuration(package, config)
+            .map(move |()| {
+                Some(EitherIter::Left(configured_package_combinations(
+                    package, config, groups,
+                )))
+            })
+            .map_err(|error: Error| eprintln!("Configuration checks failed!\nError: {error}"))
```

Comment:
the first time I see "error swallowing" in Rust
I hope it's not necessary to explain more what the problem is with this